source/common/network/address_impl.cc

```diff
@@ -57,16 +57,16 @@ bool ipFamilySupported(int domain) {
 
 Address::InstanceConstSharedPtr addressFromSockAddr(const sockaddr_storage& ss, socklen_t ss_len,
                                                     bool v6only) {
-  RELEASE_ASSERT(ss_len == 0 || ss_len >= sizeof(sa_family_t), "");
+  RELEASE_ASSERT(ss_len == 0 || static_cast<unsigned int>(ss_len) >= sizeof(sa_family_t), "");
```

Comment:
I deleted the ss_len >= 0 clause because then when the concrete type _is in fact_ unsigned we get `error: comparison of unsigned expression >= 0 is always true [-Werror=type-limits]`. I am searching today if there is a better way to handle that depending on the concrete type that `socklen_t` ends up being defined as.

Reply:
I think this is probably fine given what the code is doing.